src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/FilingsServiceTest.java

```diff
@@ -0,0 +1,80 @@
+import uk.gov.companieshouse.limitedpartnershipsapi.model.partnership.dto.DataDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.partnership.dto.LimitedPartnershipDto;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.GENERAL_PARTNER_FIELD;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.LIMITED_PARTNERSHIP_FIELD;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.LIMITED_PARTNER_FIELD;
+
+@ExtendWith(MockitoExtension.class)
+@SpringBootTest
+class FilingsServiceTest {
+
+    private static String TRANSACTION_ID = "trns123";
+    @Autowired
+    private FilingsService filingsService;
+    @MockitoBean
+    private LimitedPartnershipService limitedPartnershipService;
+    @MockitoBean
+    private GeneralPartnerService generalPartnerService;
+    @MockitoBean
+    private LimitedPartnerService limitedPartnerService;
+
+    @Test
+    void testFilingGenerationSuccess() throws ServiceException {
+        var transaction = new Transaction();
+        transaction.setId(TRANSACTION_ID);
+        when(limitedPartnershipService.getLimitedPartnership(transaction)).thenReturn(buildLimitedPartnership());
+        when(generalPartnerService.getGeneralPartnerList(transaction)).thenReturn(buildGeneralPartnerDtoList());
+        when(limitedPartnerService.getLimitedPartnerList(transaction)).thenReturn(buildLimitedPartnerDtoList());
+        FilingApi filing = filingsService.generateLimitedPartnerFilings(transaction);
+        assertNotNull(filing);
+        assertNotNull(filing.getData());
+        assertTrue(filing.getData().containsKey(LIMITED_PARTNERSHIP_FIELD));
+        assertTrue(filing.getData().containsKey(GENERAL_PARTNER_FIELD));
+        assertTrue(filing.getData().containsKey(LIMITED_PARTNER_FIELD));
+    }
+
+    @Test
+    void testFilingGenerationWithException() throws ServiceException {
+        var transaction = new Transaction();
+        transaction.setId(TRANSACTION_ID);
+        when(limitedPartnershipService.getLimitedPartnership(transaction)).thenThrow(ServiceException.class);
+        assertThrows(ServiceException.class, () -> filingsService.generateLimitedPartnerFilings(transaction));
+    }
+
+    private LimitedPartnershipDto buildLimitedPartnership() {
+       var limitedPartnershipDto = new LimitedPartnershipDto();
+       var dataDto = new DataDto();
+       limitedPartnershipDto.setData(dataDto);
+       return limitedPartnershipDto;
+    }
+
+    private List<GeneralPartnerDto> buildGeneralPartnerDtoList()
+    {
+        return null;
+    }
+
+    private List<LimitedPartnerDto> buildLimitedPartnerDtoList()
+    {
+        return null;
+    }
```

Comment:
Is this a valid test. Wouldn't an empty list be returned if no records were found?

Reply:
Need to update these.

Reply:
Removed them